### ISSUE TITLE
Expose documentation tool for godot rust projects

### ISSUE THREAD

**Issue by astrale-sharp** [godot-rust/gdnative-book#42]
Hi there, i have found a very interesting tool that i think we should mention in the book.
[gdnative-doc](https://docs.rs/gdnative-doc/0.0.5/gdnative_doc/) parses the documentation in the rust code and generate the documentation in markdown (that can be shipped with an addon you're developping for instance)
It can also take your doc test in ```gdscript and put them into a test_file so they can be run by [Gut](https://github.com/bitwes/Gut).
Its also flawlessly integrated into cargo

It is used in [a project i contributed to](https://github.com/MatejSloboda/Dijkstra_map_for_Godot)

It would be pertinent to look at
1.  [the build.rs file](https://github.com/MatejSloboda/Dijkstra_map_for_Godot/blob/master/dijkstra-map-gd/build.rs)
2. [the structure exported to godot](https://github.com/MatejSloboda/Dijkstra_map_for_Godot/blob/master/dijkstra-map-gd/src/lib.rs)
3. [the generated doc](https://github.com/MatejSloboda/Dijkstra_map_for_Godot/blob/master/addons/dijkstra-map/doc/)
4. [the generated Test](https://github.com/MatejSloboda/Dijkstra_map_for_Godot/blob/master/Tests/unit/DijkstraMap.gd)

This could all be part of a section about developping addons in godot-rust for instance (which i would be interested in writing) what do you think?

**Comment by astrale-sharp** [comment]
by the way, it make me think it could be used to implement an automatic test from godot features using this and the command line of the Gut addon, just an idea

**Comment by Bromeon** [comment]
Generally sounds like a good idea -- there are some other integrations worth mentioning in the future, e.g. godot-egui.

However, there are some caveats:
* Some tools are very specialized for the author's use case, and open-sourced so that others could potentially use it, but not to build a library/tool as primary goal.
* Early prototype of tools tend to change a lot. If we add examples, they may quickly become outdated. I expect that tool authors are not actively updating the godot-rust book to reflect their changes all the time. And I don't see the godot-rust team chasing external APIs, either.

So maybe it makes sense to have a section `Third-party integrations` (maybe after `9. Advanced Guides`). This section _mentions_ tools, libraries and other integrations, contains 2-3 high-level sentences about the purpose as well as a link, but does _not_ have code or other examples. It is then the responsibility of said integrations to provide a motivating, up-to-date description on their main page (website or GitHub repo).

Even then, there would still need to be some basic curation; projects that haven't been updated in a while could be removed, dead links would need to be updated, etc.

I am also thinking we could do something similar for games built with godot-rust.

What do you and others think?

**Comment by astrale-sharp** [comment]
I think it makes a lot of sense to put this and the like in a `Third-party integrations` and also to have games made in godot rust somewhere `Made with godot-rust`

I'd be interested to open a PR on the subject

**Comment by Bromeon** [comment]
Glad to hear that! Maybe let's wait for some more input before rushing to implementation 🙂

**Comment by jacobsky** [comment]
I mentioned this in the discord, but I'm all for a "Third Party Integrations" section, so long as we very explicitly mention that these are "as-is" and that they are intended to act as a starting point, not necessarily as anything that we explicitly support. If we go this route.

**Comment by astrale-sharp** [comment]
Yes ofc! I'm just showing my enthusiasm !
A little list of the task at hands that I will keep updated here to keep things organized :
- [ ] Modify src/SUMMARY.gd : add a third Item (Third-party integrations) to Advanced Guides
- [ ] Write a little summary/ explanation of the goal of this new part of the book in third_party_integrations.md 
- [ ] List some third-party integrations mentioned on discord or elsewhere namely :
- gdnative-doc ? (mentioned upper)
- godotEgui (mentioned below)
- project-utils (mentioned below)

Please ping me (or maybe open an issue) if you know of more third-party integrations that should be included.
This is just a proposition, if you think something should be changed, let's discuss it!

**Comment by jacobsky** [comment]
@astrale-sharp That plan looks good to me for now. Third-party integrations can be in either Advanced Guides or as it's own sections. I don't think we need to worry about it too much at the moment.

A third party integration that I am working with another member of the community is [GodotEgui](https://crates.io/crates/godot_egui) and while there's a fair bit of documentation on the [github repo's readme](https://github.com/setzer22/godot-egui/), if you are interested in creating a section we'd always appreciate it. Otherwise, I might take a day to add the section as well.

One project that got a little delayed was the [project-utils](https://github.com/godot-rust/project-utils) which is a bit old, but could be useful to have a guide for configuring it.
For my part, I'm in the process of creating an Addon that can help to integrate it all a bit better into Godot.

**Comment by astrale-sharp** [comment]
Taken into account by editing the previous comment ! 

I'd be happy to write a section about GodotEgui.

I was of the opinion of @Bromeon that the tutorial part should be the responsibility of the others projects. What do you think?

**Comment by Bromeon** [comment]
We are adding the empty pages for third-party projects in https://github.com/godot-rust/book/pull/46.
As soon as it's merged, you are welcome to add some content!

godot-egui would fit into the _Tools and Integrations_ part. You are right, we should leave the tutorial up to the other projects. Maybe just write 2-3 high-level sentences what the project is about, and if possible add a screenshot 🙂

Could you reuse #49 for this, so we don't have dozens of empty PRs?
(or #47 if you want me to re-open it)

**Comment by jacobsky** [comment]
I agree with the sentiment that tutorials are best left to the recipes section or the relevant projects. As long as we provide a link to the main repository most users should be able to find the relevant docs.
